### PR TITLE
Fix -> Android native exceptions

### PR DESCRIPTION
**Pull request by aykutuludag**
com.gdelataillade.alarm.alarm.AlarmService.onStartCommand java.lang.NullPointerException - Attempt to invoke virtual method 'int java.lang.String.length()' on a null object reference

### CODE REVIEW

**Comment by aykutuludag** [comment]
https://github.com/gdelataillade/alarm/issues/248

File: android/src/main/kotlin/com/gdelataillade/alarm/alarm/BootReceiver.kt [hunk 1]
```diff
@@ -33,6 +33,8 @@ class BootReceiver : BroadcastReceiver() {
 
         for (alarm in storedAlarms) {
             // Create the arguments for the MethodCall
+            val notificationSettings = alarm.notificationSettings ?: NotificationSettings.default()
+
```
Comment (gdelataillade) [inline android/src/main/kotlin/com/gdelataillade/alarm/alarm/BootReceiver.kt:37]: It looks like you added this new variable but you're not using it.

File: android/src/main/kotlin/com/gdelataillade/alarm/alarm/AlarmService.kt [hunk 2]
```diff
@@ -67,13 +67,17 @@ class AlarmService : Service() {
...
-        val jsonObject = JSONObject(notificationSettingsJson)
-        val map: MutableMap<String, Any> = mutableMapOf()
-        jsonObject.keys().forEach { key ->
-            map[key] = jsonObject.get(key)
+        val notificationSettings = if (!notificationSettingsJson.isNullOrEmpty()) {
+            val jsonObject = JSONObject(notificationSettingsJson)
+            val map: MutableMap<String, Any> = mutableMapOf()
+            jsonObject.keys().forEach { key -> map[key] = jsonObject.get(key) }
+            NotificationSettings.fromJson(map)
+        } else {
```
Comment (gdelataillade) [inline android/src/main/kotlin/com/gdelataillade/alarm/alarm/AlarmService.kt:77]: Maybe remove the else block ? Well notificationSettingsJson is never supposed to be null or empty but if it's the case then we don't want to show a notification.

Comment (aykutuludag) [reply android/src/main/kotlin/com/gdelataillade/alarm/alarm/AlarmService.kt:77]: It is empty when users switched from old alarm library versions if I understand correctly.

Comment (gdelataillade) [reply android/src/main/kotlin/com/gdelataillade/alarm/alarm/AlarmService.kt:77]: you're right. I realize I have to add backward compatibility with older models in native code. Don't worry I'll add it myself. You can close this PR. Thanks for pointing this issue out.

**Pull request closed without merging**